gym/spaces/space.py

```diff
@@ -26,41 +42,41 @@ def __init__(self, shape=None, dtype=None, seed=None):
             self.seed(seed)
 
     @property
-    def np_random(self):
+    def np_random(self) -> np.random.RandomState:
         """Lazily seed the rng since this is expensive and only needed if
         sampling from this space.
         """
         if self._np_random is None:
             self.seed()
 
-        return self._np_random
+        return self._np_random  # type: ignore  ## self.seed() call guarantees right type.
```

Comment:
Wouldn't it pass type checks if `_np_random` gets added as a field to the class? This feels like something that shouldn't require disabling the checker (although it is a little bit unusual)

Reply:
The problem is to narrow down the type after self.seed() is called - as it is self._np_random is Optional[np.random.RandomState], and pyright doesn't understand that if it's None, than it's made not-None by self.seed() call.